### PR TITLE
[AIR][Numpy] Update image example to use numpy API

### PR DESCRIPTION
**Pull request by jiaodong**
## Why are these changes needed?

Current one uses pandas that's not recommended for image workloads (not tensor centric)

Only change is made by manually pasting json cell for this part, all other changes seems triggered by some default jupyer behavior, but no outputs are manually modified.

Screen Shot 2022-10-21 at 4 38 11 PM

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by matthewdeng** [comment]
@xwjiang2010

**Comment by amogkam** [comment]
why do we need the stacks? isn't it just this 

```
images = np.array([image.numpy() for image, _ in batch])
labels = np.array([label for _, label in batch])
```

**Comment by jiaodong** [comment]
> why do we need the stacks? isn't it just this
> 
> ```
> images = np.array([image.numpy() for image, _ in batch])
> labels = np.array([label for _, label in batch])
> ```

oh this works too. I was thinking of adding outer dim but np.array is equivalent for this case.

File: doc/source/ray-air/examples/torch_image_example.ipynb [hunk 1]
```diff
@@ -158,7 +158,7 @@
             "source": [
                 "{py:class}`SimpleTorchDatasource <ray.data.datasource.SimpleTorchDatasource>` doesn't parallelize reads, so you shouldn't use it with larger datasets.\n",
                 "\n",
-                "Next, let's represent our data using pandas dataframes instead of tuples. This lets us call {py:meth}`Dataset.iter_torch_batches <ray.data.Dataset.iter_torch_batches>` later in the tutorial."
+                "Next, let's represent our data using dictionary of ndarrays instead of tuples. This lets us call {py:meth}`Dataset.iter_torch_batches <ray.data.Dataset.iter_torch_batches>` later in the tutorial."
```
Comment (clarkzinzow) [inline doc/source/ray-air/examples/torch_image_example.ipynb:161]: Nit:
```suggestion
                "Next, let's represent our data using a dictionary of ndarrays instead of tuples. This lets us call {py:meth}`Dataset.iter_torch_batches <ray.data.Dataset.iter_torch_batches>` later in the tutorial."
```

File: doc/source/ray-air/examples/torch_image_example.ipynb [hunk 2]
```diff
@@ -205,20 +205,20 @@
...
                 "import torch\n",
                 "\n",
                 "\n",
-                "def convert_batch_to_pandas(batch: Tuple[torch.Tensor, int]) -> pd.DataFrame:\n",
-                "    images = [image.numpy() for image, _ in batch]\n",
-                "    labels = [label for _, label in batch]\n",
-                "    return pd.DataFrame({\"image\": images, \"label\": labels})\n",
+                "def convert_batch_to_numpy(batch: Tuple[torch.Tensor, int]) -> Dict[str, np.ndarray]:\n",
+                "    images = np.array([image.numpy() for image, _ in batch])\n",
+                "    labels = np.array([np.asarray(label) for _, label in batch])\n",
```
Comment (clarkzinzow) [inline doc/source/ray-air/examples/torch_image_example.ipynb:215]: Hmm why is this nesting necessary for the label? Shouldn't this work?
```suggestion
                "    labels = np.array([label for _, label in batch])\n",
```

**Review by clarkzinzow — APPROVED** [review]
LGTM!

**Review by xwjiang2010 — APPROVED** [review]
Thanks Jiao!

**Review by richardliaw — APPROVED** [review]
